galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt

```diff
@@ -2041,36 +2043,42 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
     }
   }
 
+  @Suppress("UNCHECKED_CAST")
   fun setValueID(id: Int) {
-    TODO()
-    /*
-    var result: Any? = null
-
-    try {
-      while (true) {
-        try {
-          transaction {
-            exec("SELECT " + list!!.getColumn(0).column!! + " FROM "
-                    + evalListTable() + " WHERE ID = " + id) {
-              result = if (it.next()) {
-                it.getObject(1)
-              } else {
-                null
-              }
-            }
-          }
-          break
-        } catch (e: SQLException) {
-        } catch (error: Error) {
-        } catch (rte: RuntimeException) {
-        }
+    val result = try {
+      transaction {
+        val table= evalListTable()
+        val idColumn = table.columns.find { it.name == "ID" } as Column<Int>
+        val firstRecord = table.slice(table.resolveColumn(list!!.getColumn(0).column!!)).select {
```

Comment:
if VListColumn contain an val cloumn_ de type exposed.column (sarrah pull request) so you don't need to fetch column here !